chain/index/events.go

```diff
@@ -0,0 +1,560 @@
+			}
+
+			log.Infof("successfully recomputed tipset state and loaded events amt for message %s", ems[i].msg.Cid())
+		}
+
+		ems[i].evs = make([]types.Event, eventsArr.Len())
+		var evt types.Event
+		err = eventsArr.ForEach(ctx, func(u uint64, deferred *cbg.Deferred) error {
+			if u > math.MaxInt {
+				return xerrors.Errorf("too many events")
+			}
+			if err := evt.UnmarshalCBOR(bytes.NewReader(deferred.Raw)); err != nil {
+				return err
+			}
+
+			cpy := evt
+			ems[i].evs[int(u)] = cpy
+			return nil
+		})
+
+		if err != nil {
+			return nil, xerrors.Errorf("failed to iterate over events for message %d: %w", i, err)
+		}
+
+	}
+
+	return ems, nil
+}
+
+// checkTipsetIndexedStatus verifies if a specific tipset is indexed based on the EventFilter.
+// It returns nil if the tipset is indexed, ErrNotFound if it's not indexed or not specified,
+func (si *SqliteIndexer) checkTipsetIndexedStatus(ctx context.Context, f *EventFilter) error {
+	var tipsetKeyCid []byte
+	var err error
+
+	// Determine the tipset to check based on the filter
+	switch {
+	case f.TipsetCid != cid.Undef:
+		tipsetKeyCid = f.TipsetCid.Bytes()
+	case f.MinHeight >= 0 && f.MinHeight == f.MaxHeight:
+		tipsetKeyCid, err = si.getTipsetKeyCidByHeight(ctx, f.MinHeight)
+		if err != nil {
+			if err == ErrNotFound {
+				// this means that this is a null round and there exist no events for this epoch
+				return nil
+			}
+
+			return xerrors.Errorf("failed to get tipset key cid by height: %w", err)
+		}
+	default:
+		// Filter doesn't specify a specific tipset
+		return nil
+	}
+
+	// If we couldn't determine a specific tipset, return ErrNotFound
+	if tipsetKeyCid == nil {
+		return ErrNotFound
+	}
+
+	// Check if the determined tipset is indexed
+	if exists, err := si.isTipsetIndexed(ctx, tipsetKeyCid); err != nil {
+		return xerrors.Errorf("failed to check if tipset is indexed: %w", err)
+	} else if exists {
+		return nil // Tipset is indexed
+	}
+
+	return ErrNotFound // Tipset is not indexed
+}
+
+// getTipsetKeyCidByHeight retrieves the tipset key CID for a given height.
+func (si *SqliteIndexer) getTipsetKeyCidByHeight(ctx context.Context, height abi.ChainEpoch) ([]byte, error) {
+	ts, err := si.cs.GetTipsetByHeight(ctx, height, nil, false)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to get tipset by height: %w", err)
+	}
+	if ts == nil {
+		return nil, xerrors.Errorf("tipset is nil for height: %d", height)
+	}
+
+	if ts.Height() != height {
+		// this means that this is a null round
+		return nil, ErrNotFound
+	}
+
+	return toTipsetKeyCidBytes(ts)
+}
+
+// GetEventsForFilter returns matching events for the given filter
+// Returns nil, nil if the filter has no matching events
+// Returns nil, ErrNotFound if the filter has no matching events and the tipset is not indexed
+// Returns nil, err for all other errors
+func (si *SqliteIndexer) GetEventsForFilter(ctx context.Context, f *EventFilter, excludeReverted bool) ([]*CollectedEvent, error) {
```

Comment:
```suggestion
func (si *SqliteIndexer) GetEventsForFilter(ctx context.Context, f *EventFilter) ([]*CollectedEvent, error) {
```

I'm pretty sure you can just use `f` for this since it's always the same, all the way down into `makePrefillFilterQuery` where you still have `f` you could do it right where you're querying by tipset. If you ask for a tipset, then don't exclude reverted, otherwise, exclude reverted.

Reply:
This is done.